msal/application.py

```diff
@@ -585,6 +612,9 @@ def acquire_token_by_device_flow(self, flow, **kwargs):
                     # 2018-10-4 Hack:
                     # during transition period,
                     # service seemingly need both device_code and code parameter.
+                headers={'client-request-id': _get_new_correlation_id(),
```

Comment:
The following topic is more about Device Flow.

Shall we consider to let the initial Device Flow call and all its subsequent polling to have same `client-request-id`? It is easy to implement too. We just store the first id as `_client_request_id` in the `flow` object, and then reuse it here. Something like this:
```suggestion
                headers={
                    'client-request-id': flow.get("_client_request_id", _get_new_correlation_id()),
```

Reply:
I'm not sure how server generates device codes - I think it's a safer bet to use UUIDs to avoid collisions.

Reply:
I'm not sure how server generates device code. I would rather use UUID as, as their is less chance of collision (and this is what we use for every other flow, don't see reason to change it here)